build_tools/waf_content.py

```diff
@@ -138,9 +137,11 @@ def sig_explicit_deps(self):
             proto_module_sigs[module] = mod_m.digest()
 
         deps = old_sig_explicit_deps(self)
+
         m = hashlib.md5()
         m.update(proto_module_sigs[module])
-        m.update(deps)
+        if deps != None:
+            m.update(deps)
```

Comment:
> Apart from one or two bugs, I think it looks mostly good. Some questions:
> 
> Why the need to specify the install path when we already pass it to waf with "--prefix=$DYNAMO_HOME" ? I'm guessing you've built the libraries one by one, but forgot to set the prefix? ( `waf --prefix=${DYNAMO_HOME} configure`)

I'm pretty sure that waf has changed something in their API for how static libs are treated in the install step, because if I don't specify the install path the static libs doesn't get installed in dynamo home when invoking the build.py script. This is what the WAF book says:

"Programs and shared libraries are installed by default. To disable the installation, set None."
https://waf.io/book/#:~:text=Programs%20and%20shared%20libraries%20are%20installed%20by%20default.%20To%20disable%20the%20installation%2C%20set%20None.

> 
> Why the need to change to "w" as opposed to keeping the "wb" to preserve the data specified? Not a huge deal, but I've had so many issues due to "text" formatting in the past, I'm a bit waery about it.

Hm, good question. I'll have to take a look at some of the cases. We could perhaps use 'wb', but p3 is a lot more strict when it comes to dealing with strings so I'm guessing I took a shortcut with those maybe. 

> 
> About migration: Can we add "syntax = "proto2";" already, with python2 or do we need to wait to update the Rive/Spine extensions' .proto files?
Adding the syntax="proto2"; is not strictly necessary, protoc will default to version 2 anyway. Adding the syntax will only prevent a warning from showing.